worlds/witness/test/__init__.py

```diff
@@ -0,0 +1,149 @@
+from test.bases import WorldTestBase
+from test.general import gen_steps, setup_multiworld
+from test.multiworld.test_multiworlds import MultiworldTestBase
+from typing import Any, ClassVar, Dict, Iterable, List, Mapping, Union, cast
+
+from BaseClasses import CollectionState, Entrance, Item, Location, Region
+
+from .. import WitnessWorld
+
+
+class WitnessTestBase(WorldTestBase):
+    game = "The Witness"
+    player: ClassVar[int] = 1
+
+    world: WitnessWorld
+
+    def can_beat_game_with_items(self, items: Iterable[Item]) -> bool:
+        """
+        Check that the items listed are enough to beat the game.
+        """
+
+        state = CollectionState(self.multiworld)
+        for item in items:
+            state.collect(item)
+        return state.multiworld.can_beat_game(state)
+
+    def assert_dependency_on_event_item(self, spot: Union[Location, Region, Entrance], item_name: str) -> None:
```

Comment:
Does it really make sense to use events in tests if the thing players collect are items, not events? This sounds like testing an implementation detail rather than the logic.

Reply:
The implementation detail is what I want to test. These events have disappeared in the past, but in a way where things didn't actually break.